Fix: Undefined array key on Asset import

eduardomozart opened the pull request:
<!--

Dear GLPI developer.

BEFORE SUBMITTING YOUR PULL REQUEST, please make sure to read and follow these steps:

* We don't support community plugins. Contact directly their authors, or use the community forum : http://forum.glpi-project.org.
* For feature requests or enhancements, use the suggest dedicated site (http://suggest.glpi-project.org). We check it very often.
* We prefer to keep this tracker in ENGLISH. If you want support in your language, the community forum (http://forum.glpi-project.org) is the best place.
* Evolutions and features should target the `main` branch and should be discussed in an issue before submitting a PR.
* Bug fixes should target the latest stable release branch (usually the default branch).
* Please use the below template.

For more information, please check contributing guide:
https://github.com/glpi-project/glpi/blob/main/CONTRIBUTING.md

The GLPI team.
-->

## Checklist before requesting a review

*Please delete options that are not relevant.*

- [X] I have read the CONTRIBUTING document.
- [ ] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes # (issue number, if applicable)
- Here is a brief description of what this PR does

## Screenshots (if appropriate):

The error "Undefined array key" is thrown on GLPI-Agent 1.12 when importing Inventory from a Windows machine when modifying the "Computer update (by serial + uuid)" from "Link if possible" (default) to "Link if possible, otherwise imports declined".

<img width="1112" alt="image" src="https://github.com/user-attachments/assets/344bd7dc-c367-4243-bed7-1e82854d7705" />

I did a ``error_log(print_r($datarules, true))`` before the if call and it returns the following:

```
(
    [rules_id] => 247
    [_ruleid] => 247
)
```

This ID reflects the rule ``Computer update (by serial + uuid)`` which refers to the Rule where it stopped and the asset hasn't been imported. Notice that there's no ``action`` key on the array. This PR do the ``action`` validation only if it's set.

The machine doesn't exist on the ``glpi_computers`` DB (it existed before but has been deleted). The inventory file of this machine is in attachment.

[server02-with-glpi-agent.json](https://github.com/user-attachments/files/18661894/server02-with-glpi-agent.json)

stonebuzz commented:
Hi @eduardomozart,

I successfully completed the first import without any issues or `Undefined array key`. I then adjusted the rule accordingly.

The second import also ran without any errors.

Could you provide me with an export of your computer rules?

![image](https://github.com/user-attachments/assets/9716b313-4d05-4d62-bb53-ccd195556ba4)

trasher commented:
A test case to reproduce is needed

eduardomozart commented:
Hello @trasher and @stonebuzz, I've updated the inventory file and exported the rules XML file as requested and was able to reproduce the issue. Please make sure to attempt to import the file with the Debug mode enabled. It states "File has been successfully imported" but I think that for "Import denied (no log)" it should also return as refused because it can confuse the user, as in fact the Computer asset hasn't been imported.

<img width="1435" alt="image" src="https://github.com/user-attachments/assets/b2b599fc-5cc5-4d48-a963-c476c9796c62" />

stonebuzz commented:
Always nothing on my side

![image](https://github.com/user-attachments/assets/1e5cb527-5555-412d-b5b6-62f55ea17d90)

![image](https://github.com/user-attachments/assets/234e1d4a-e6c2-4608-8141-4895ccaa7b84)

stonebuzz commented:
Could you try again by resetting your rules?

eduardomozart commented:
If I reset the rules it should work as expected since it happens only when the rule ``Computer update (by serial + uuid)`` is set to ``Link if possible, otherwise import denied``. Did the asset has been created for you? I deleted it without keeping associated devices, changed the rule and it hasn't been imported for me at all.

stonebuzz commented:
hi @eduardomozart 

still nothing on my side

![image](https://github.com/user-attachments/assets/6e151f66-95c8-4746-9bd6-a464e93f3234)

![image](https://github.com/user-attachments/assets/a7575fc9-d331-4e0b-9d29-c498d951194b)


Are you reproducing with the latest stable and nightly versions of GLPI?

Best regards

eduardomozart commented:
Hello,
I can confirm that on GLPI 10.0.18 (Stable) this issue doesn't occurs. I believe I was using GLPI 10.0.18 when importing that asset.
